node/src/components/storage.rs

```diff
@@ -307,6 +308,22 @@ impl Storage {
         drop(cursor);
         drop(block_txn);
 
+        // Check the integrity of the block body database.
+        check_block_body_db(&env, &block_body_db).unwrap_or_else(|error| {
+            panic!(
+                "Error in the integrity check of the block_body database: {:?}",
+                error
+            )
+        });
+
+        // Check the integrity of the block metadata database.
+        check_block_metadata_db(&env, &block_metadata_db).unwrap_or_else(|error| {
+            panic!(
+                "Error in the integrity check of the block_metadata database: {:?}",
+                error
+            )
+        });
+
```

Comment:
Can you extract the indexing above to a method as well? Thank you.

Reply:
Do you mean the block header re-indexing, if so, then I'd rather leave that where it is and now disturb Marc's rewrite of the storage component.